### PR TITLE
Merge SixLabors.Core sources into ImageSharp [first attempt]

### LINKED ISSUE #1002: Externalize ImageSharp.Drawing
Prerequisite: #967

----

**UPDATE**

### Final repository & package structure
image

### Refactor steps
I think the best would be to follow the order I suggest. Please let me know if something is missing!

- [ ] Copy all contents of *SixLabors.Core* into the *SixLabors.ImageSharp* repository (test and product code)
- [x] Create a new, 100% self-contained repository for ImageSharp.Drawing, copy the following content into the repository:
    - Current Drawing code:
        - Product code
        - Test code
        - Benchmark code (!!!)
    - Test images used by Drawing tests
    - Reference images used by Drawing tests
    - Product and test code of `SixLabors.Shapes`
- [x] Configure CI for the ImageSharp.Drawing repository
- [ ] Update *ImageSharp.Web* (dogfeeding)

The steps above will also touch #868. If we want to completely resolve that issue, we shall also:
- [ ] Clean-up history of binary artifacts in *SixLabors.ImageSharp* master
- [ ] Copy all reference images to *SixLabors.ImageSharp*

We shall consider introducing GitHub workflows while implementing these steps, but I need more help if we want to do so.

### PR DESCRIPTION
**Pull request by antonfirsov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Merge sources and tests from SixLabors.Core, resolve #1002.
- Namespace structure were left untouched.
- The new source structure follows namespace hierarchy in `src`. 
- SixLabors.Core tests have been grouped under their own directory
- Most memory-intensive `ArrayPoolMemoryAllocatorTests` have been disabled temporarily. The safest way to run such tests would be arcane's [RemoteExecutor](https://github.com/dotnet/runtime/blob/master/docs/project/writing-tests.md#remoteexecutor)

*Do not squash-merge!*

### CODE REVIEW

**Comment by antonfirsov** [comment]
@JimBobSquarePants if we move the types, we should also do `SixLabors.Shapes` -> `SixLabors.Drawing.Shapes` in the drawing project for consistency. (cc @tocsoft)

I don't have a strong opinion apart from the fact that our decision should be user and business-focused now. So the question is: Does this worth the breaking changes (again) in `beta7` -> `RC1`? Wouldn't it discourage users from adapting RC1?

File: src/ImageSharp/ImageSharp.csproj.DotSettings [hunk 1]
```diff
@@ -2,6 +2,12 @@
 	<s:Boolean x:Key="/Default/CodeInspection/NamespaceProvider/NamespaceFoldersToSkip/=color/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/CodeInspection/NamespaceProvider/NamespaceFoldersToSkip/=common/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/CodeInspection/NamespaceProvider/NamespaceFoldersToSkip/=common_005Cexceptions/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/CodeInspection/NamespaceProvider/NamespaceFoldersToSkip/=imagesharp_005Cpixelformats_005Cpixelimplementations/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/CodeInspection/NamespaceProvider/NamespaceFoldersToSkip/=imagesharp_005Cpixelformats_005Cutils/@EntryIndexedValue">True</s:Boolean>
```
[the remark this answers is not in the document]
Comment (antonfirsov) [reply src/ImageSharp/ImageSharp.csproj.DotSettings]: The other "toy IDE" 😆, Rider is also using this, and it's somewhat faster. 
Still unsure what will be my IDE of choice in long term, but for now I'm pretty much addicted to JetBrains features, so I need `.DotSettings`.

"NamespaceProvider" is one of the features I really like btw.

File: tests/ImageSharp.Tests/_SixLabors.Core/Helpers/DebugGuardTests.cs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+namespace SixLabors.Helpers.Tests
+{
+    public class DebugGuardTests
+    {
+        private class Foo
+        {
+        }
+
+        [Fact]
+        public void AllStaticMethodsOnOnDebugGuardHaveDEBUGConditional()
```
[the remark this answers is not in the document]
Comment (antonfirsov) [reply tests/ImageSharp.Tests/_SixLabors.Core/Helpers/DebugGuardTests.cs]: Previously we used to test them in Core. If you setup an action in SharedInfrastructure, I'm happy to delete them here.

**Comment by JimBobSquarePants** [comment]
I agree it should be user focused. That’s why I’m opting for creating the long term expected result over immediate upgrade. 

I don’t think it’s too painful an upgrade anyway. The reference will remain and the user will be able to bulk delete the old namespace.

**Comment by antonfirsov** [comment]
> That’s why I’m opting for creating the long term expected result over immediate upgrade.

OK. Just to double check: this means that you also agree with the changes in Drawing, right? That one will have a bigger impact.

**Comment by tocsoft** [comment]
@anton changing the namespaces seems fine as an idea to me but I agree with @JimBobSquarePants and we should just totally drop the `Primitives` namespace part entirely and roll the types directly into the `SixLabors.ImageSharp` namespace. `Memory` on the other hand is more specialised and as people don't/wont be touching them regularly then `SixLabors.ImageSharp.Memory` makes sense for them. 

re: drawing & shapes not sure exactly how to cut that namespace but as that project is not a target for our RC we can worry about cleaning us the structure over there at a later date.

**Comment by antonfirsov** [comment]
Closing in favor of #1087.

**Pull request closed without merging**